Rebuild `EmptyState` component with layout primitives

yurm04 opened the pull request:
Fixes [#7564](https://github.com/Shopify/polaris/issues/7564)

- Remove max-width from AlphaStack
- Rebuild EmptyState with layout components

DEPENDS ON #7842. Merge that in first then bring `layout-rebuild-batch-2` up to date with `main` and then merge in those changes into this branch

### [✅ RESOLVED] UX Review
There is a discrepancy in the size of the image container. The update `EmptyState` component image container  is 5px smaller in height than the production one. The comparison below shows the production version (first) vs the update component (second). This results in about 5px additional whitespace in the prod version. This different is due to the production version relying on a flex property on the image container that we don't currently implement. Is this difference acceptable or should we adjust the image container size somehow?

<details>
    <summary>Compare image container whitespace before and after</summary>

https://user-images.githubusercontent.com/4642404/204934759-3363b103-5118-4335-ab87-79edaa5d0634.mov

Production image size 
<img width="1071" alt="prod-size" src="https://user-images.githubusercontent.com/4642404/204935557-dc3ea1c8-778f-4644-8c2b-62d524b2f03b.png">

Updated ima
<img width="1071" alt="updated-size" src="https://user-images.githubusercontent.com/4642404/204935620-be70ec16-0ece-432e-9105-64e94ba2525c.png">


</details>

yurm04 committed:
remove max-width from AlphaStack

yurm04 committed:
replaced with layout components

yurm04 committed:
add responsive to maxwidth on Box

kyledurand committed:
[ResourceItem] Use layout components (#7784)

Closes: https://github.com/Shopify/polaris/issues/7580

This slightly moves the checkbox handle 2px to the left. This means it
now lines up with our 20px padding instead of 22px from the card edge.
Also improves vertical alignment of the checkboxes and bulk select on xs
screens

Before | After
---|---
![image](https://screenshot.click/24-12-z59dz-bpdqu.png) |
![image](https://screenshot.click/24-12-viiqs-efiaf.png)

The only CSS that remains is for interactive elements, interactive
behavior (hover, focus), and transitions (which are being removed in
https://github.com/Shopify/polaris/pull/7290/files).

kyledurand committed:
[ResourceItem] Use layout components (#7784)

Closes: https://github.com/Shopify/polaris/issues/7580

This slightly moves the checkbox handle 2px to the left. This means it
now lines up with our 20px padding instead of 22px from the card edge.
Also improves vertical alignment of the checkboxes and bulk select on xs
screens

Before | After
---|---
![image](https://screenshot.click/24-12-z59dz-bpdqu.png) |
![image](https://screenshot.click/24-12-viiqs-efiaf.png)

The only CSS that remains is for interactive elements, interactive
behavior (hover, focus), and transitions (which are being removed in
https://github.com/Shopify/polaris/pull/7290/files).

yurm04 committed:
updated responsive props

yurm04 committed:
add getResponsiveValues

yurm04 committed:
revert story and remove padding on actions

aveline committed:
Merge branch 'main' into layout-rebuild-batch-2

aveline committed:
[Layout] Update `ActionList` to use Layout primitives (#7805)

Fixes #7458

Co-authored-by: Lo Kim <lo.kim@shopify.com>

aveline committed:
[Layout] Update `ActionList` to use Layout primitives (#7805)

### WHY are these changes introduced?

Fixes #7458 

Co-authored-by: Lo Kim <lo.kim@shopify.com>

yurm04 committed:
remove responsive max width

yurm04 committed:
remove support for responsive values in max width Box

yurm04 committed:
remove all other styles

yurm04 commented on polaris-react/src/components/EmptyState/EmptyState.scss:
```diff
@@ -1,134 +1,8 @@
-  }
-}
-
-.withinContentContainer {
```
As far as i can tell this class should always be present since our `EmptyState` components should always be the child of a content container ex. `Card` `Modal` etc. I applied these styles by default to the `Box`es in `EmptyState` but now I'm wondering if I should have conditionally applied these styles instead just in case `EmptyState` is not used in a container parent 🤔

laurkim replied:
I think we should keep the `withinContentContainer` styling.
I don't have enough context to confidently say that there's not usage of EmptyState without it being in a parent container and I worry this may introduce visual regressions.

yurm04 replied:
went back and addressed this in the component code

yurm04 commented on polaris-react/src/components/EmptyState/EmptyState.scss:
```diff
@@ -1,134 +1,8 @@
-  }
-}
-
 .imageContained {
```
had to keep this class because we don't do responsive props in `Image`

yurm04 commented on polaris-react/src/components/EmptyState/EmptyState.tsx:
```diff
@@ -113,33 +111,39 @@ export function EmptyState({
+      </Box>
+    ) : null;
 
+  const defaultBoxStyle: BoxProps = {
```
broke these out into an object because I thought I would be overriding them based on the props. Keeping for now in case I do need to have a separate `withinContentContainer` styling

laurkim replied:
See my comment about about `withinContentContainer`.

yurm04 committed:
add class to image

yurm04 committed:
fix test

laurkim commented on polaris-react/src/components/EmptyState/tests/EmptyState.test.tsx:
```diff
@@ -8,6 +8,7 @@ import {Text} from '../../Text';
 import {UnstyledLink} from '../../UnstyledLink';
```
It's not showing up on the diff but the `Stack` import on line 6 of this file needs to be removed to resolve the build errors in ci check

laurkim commented on polaris-react/src/components/EmptyState/EmptyState.tsx:
```diff
@@ -113,33 +116,39 @@ export function EmptyState({
+    paddingBlockEnd: '16',
+  };
+
+  // new return
```
Can delete comment

laurkim commented on polaris-react/src/components/EmptyState/EmptyState.tsx:
```diff
@@ -113,33 +116,39 @@ export function EmptyState({
+    paddingInlineStart: '5',
+    paddingInlineEnd: '5',
+    paddingBlockStart: '5',
+    paddingBlockEnd: '16',
```
There's an extra 4px in the total height of empty state since we're using `--p-space-16` which is 4px more than the original spacing of `calc(3 * var(--p-space-5))`. Is this expected?

yurm04 replied:
thanks for flagging! I needed to check with @nayeobkim @sarahill about this but the TODO got deleted

aveline committed:
Merge branch 'layout-rebuild-batch-2' of https://github.com/Shopify/polaris into layout-rebuild-batch-2

yurm04 committed:
Merge branch 'layout-rebuild-batch-2' of https://github.com/Shopify/polaris into rebuilt-empty-state

laurkim committed:
[polaris.shopify.com] Run `get-props` script

yurm04 committed:
Merge branch 'layout-rebuild-batch-2' of https://github.com/Shopify/polaris into rebuilt-empty-state

chazdean committed:
[SkeletonPage] Rebuild with layout primitives (#7797)

<!--
  ☝️How to write a good PR title:
- Prefix it with [ComponentName] (if applicable), for example: [Button]
  - Start with a verb, for example: Add, Delete, Improve, Fix…
  - Give as much context as necessary and as little as possible
  - Prefix it with [WIP] while it’s a work in progress
-->

### WHY are these changes introduced?

Fixes [#7714](https://github.com/Shopify/polaris/issues/7714)

<!--
  Context about the problem that’s being addressed.
-->

### WHAT is this pull request doing?

- Refactors `SkeletonPage` and its child components to use our layout
primitives.
- Removes `max-width: 100%` property from AlphaStack
- Adds storybook examples for `SkeletonPage` with `narrowWidth` and with
`fullWidth` props


#### Issues with current PR:

   <details>
<summary>Unable to center `SkeletonPage` when `narrowWidth` prop is used
- SOLVED ✅</summary>
<table>
<tr>
<td> Before </td> <td> After </td>
</tr>
<tr>
<td> <img width="400" alt="Screenshot 2022-11-24 at 9 13 21 AM"
src="https://user-images.githubusercontent.com/59836805/203805854-4748a46a-8208-476e-9c16-d0071e4ff66b.png">
</td>
<td> <img width="400" alt="Screenshot 2022-11-24 at 9 10 25 AM"
src="https://user-images.githubusercontent.com/59836805/203806188-bd5cde58-0787-46f1-a2b3-cbaf2a228d52.png">
</td>
</tr>
<tr>
<td> <img width="400" alt="Screenshot 2022-11-24 at 9 13 42 AM"
src="https://user-images.githubusercontent.com/59836805/203806392-466df6a0-29ee-4d06-bc40-523b60b97853.png">
</td>
<td> <img width="400" alt="Screenshot 2022-11-24 at 9 11 17 AM"
src="https://user-images.githubusercontent.com/59836805/203806492-6793f6a3-591a-4cb4-9f13-46da44f64dfe.png">
</td>
</tr>
</table>
    </details>

<details>
<summary>Our current `Text` component has the opposite breakpoint
behaviour compared to how `.Title` is used in `SkeletonPage` - SOLVED
✅</summary>
      <table>
<tr>
<td> Before </td> <td> After </td>
</tr>
<tr>
<td> <img width="400" alt="before text"
src="https://user-images.githubusercontent.com/59836805/203808391-3f3bd9d2-421e-4e09-a843-e579b936d39f.gif">
</td>
<td> <img width="400" alt="before text"
src="https://user-images.githubusercontent.com/59836805/203808879-55e311ef-708b-44c9-a28e-4582c77d0992.gif">
</td>
</tr>
<tr>
<td> 

```java
.Title {
  font-weight: var(--p-font-weight-semibold);
  font-size: 24px;
  line-height: 28px;

  @media #{$p-breakpoints-md-up} {
    font-size: 20px;
  }
}
```

 </td>
<td>
    
```java
.headingXl {
  font-size: var(--p-font-size-300);
  line-height: var(--p-font-line-height-3);

  @media #{$p-breakpoints-md-up} {
    font-size: var(--p-font-size-400);
    line-height: var(--p-font-line-height-4);
  }
}
```
    
</td>
</tr>
</table>
    </details>



~~`role` prop is missing from `Box`, creating a separate issue to solve
this [here](https://github.com/Shopify/polaris/pull/7799)~~ SOLVED ✅

<!--
  Summary of the changes committed.

Before / after screenshots are appreciated for UI changes. Make sure to
include alt text that describes the screenshot.

If you include an animated gif showing your change, wrapping it in a
details tag is recommended. Gifs usually autoplay, which can cause
accessibility issues for people reviewing your PR:

    <details>
      <summary>Summary of your gif(s)</summary>
      <img src="..." alt="Description of what the gif shows">
    </details>
-->

<!-- ℹ️ Delete the following for small / trivial changes -->

### 🎩 checklist

- [ ] Tested on
[mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple
browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for
[accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted
documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md)
changes in the style guide

Co-authored-by: Lo Kim <lo.kim@shopify.com>

yurm04 committed:
remove stack from tests

yurm04 committed:
apply withinContentContainer styling

yurm04 commented on polaris-react/src/components/EmptyState/EmptyState.stories.tsx:
```diff
@@ -70,3 +70,19 @@ export function WithFullWidthLayout() {
   );
 }
+
+export function WithoutContentContainer() {
```
added this story for any cases when an empty state component is not wrapped in a content container

yurm04 commented on polaris-react/src/components/AlphaStack/AlphaStack.scss:
```diff
@@ -7,7 +7,7 @@
   --pc-stack-gap-xl: var(--pc-stack-gap-lg);
   display: flex;
-  flex-direction: column;
+  flex-direction: var(--pc-stack-order);
```
we need to set `flex-direction: column-reverser` for the `withinContentContainer` style below so adding order `AlphaStack`

aveline replied:
I would break out the update to `AlphaStack` and create a separate PR against `main`. That way `reverseOrder` is immediately available on `AlphaStack` in case others need to use it. Otherwise the change will be stuck on this branch which isn't going to get merged in for a while yet.

yurm04 replied:
broke this out into #7842

aveline commented on polaris-react/src/components/EmptyState/EmptyState.tsx:
```diff
@@ -90,56 +92,61 @@ export function EmptyState({
   const childrenMarkup = children ? (
-    <div className={styles.Content}>
-      <Text as="span" variant="bodyMd" color="subdued">
+    <Box>
```
I've been thinking about whether there's value in replacing a simple `div` with `Box` with no props. Curious to hear your thoughts.

yurm04 replied:
makes sense to me! I think the only one we could argue for keeping would be the `Box` that wraps the image since it does add an extra 5px height as the container. However, UX has signed off on that being removed so we don't need any of them at this point!

sarahill reviewed (approved):
✅

yurm04 committed:
add delay to story

yurm04 committed:
remove Boxes with no styling

yurm04 commented on polaris-react/src/components/EmptyState/EmptyState.stories.tsx:
```diff
@@ -4,6 +4,11 @@ import {Card, EmptyState, Link} from '@shopify/polaris';
+  parameters: {
+    // Sets a delay for EmptyState because there is a slight layout shift
+    // when first rendered.
+    chromatic: {delay: 300},
```
trying this out in response to the height shift difference in the chromatic test

yurm04 committed:
revert alphastack

yurm04 committed:
reset alphastack

yurm04 committed:
reset alphastack

yurm04 committed:
Merge https://github.com/Shopify/polaris into layout-rebuild-batch-2

yurm04 committed:
Merge branch 'layout-rebuild-batch-2' in

yurm04 committed:
uped chromatic delay

yurm04 committed:
uped chromatic

yurm04 committed:
uped chromatic

yurm04 committed:
Merge branch 'layout-rebuild-batch-2' into rebuilt-empty-state

yurm04 committed:
revert to current state

yurm04 committed:
remove chromatic delay

chazdean reviewed (approved):
LGTM 🚀 🎩
